src/main/java/org/jenkinsci/test/acceptance/po/PluginManager.java

```diff
@@ -264,7 +265,10 @@ public boolean installPlugins(final PluginSpec... specs) throws UnableToResolveD
         }
 
         // Jenkins will be restarted if necessary
-        new UpdateCenter(jenkins).waitForInstallationToComplete(specs);
+        boolean hasBeenRestarted = new UpdateCenter(jenkins).waitForInstallationToComplete(specs);
+        if (!hasBeenRestarted && specs.length > 0) {
+            jenkins.getLogger("all").waitForLogged(Pattern.compile("Completed installation of .*"), 1000);
```

Comment:
This surely is an interesting approach, but note this can match previously installed plugin or a successful installation of a dependency of the required plugin. How about waiting for this to be logged for every explicitly requires plugin that was yet `NOT_INSTALLED`?

Reply:
I'm not sure if I can follow you here, my knowledge about that part in the code is quite limited. From the console log I see that the tests currently start before 
```
master63023|2020-03-11 23:54:13.683+0000 [id=78]	INFO	jenkins.InitReactorRunner$1#onAttained: Started initialization
master63023|2020-03-11 23:54:13.748+0000 [id=85]	INFO	jenkins.InitReactorRunner$1#onAttained: Listed all plugins
master63023|2020-03-11 23:54:13.748+0000 [id=85]	INFO	jenkins.InitReactorRunner$1#onAttained: Prepared all plugins
master63023|2020-03-11 23:54:13.785+0000 [id=81]	INFO	jenkins.InitReactorRunner$1#onAttained: Started all plugins
master63023|2020-03-11 23:54:13.799+0000 [id=81]	INFO	jenkins.InitReactorRunner$1#onAttained: Augmented all extensions
master63023|2020-03-11 23:54:13.800+0000 [id=84]	INFO	jenkins.InitReactorRunner$1#onAttained: Loaded all jobs
master63023|2020-03-11 23:54:13.816+0000 [id=79]	INFO	jenkins.InitReactorRunner$1#onAttained: Completed initialization
master63023|2020-03-11 23:54:14.075+0000 [id=75]	INFO	h.m.UpdateCenter$CompleteBatchJob#run: Completed installation of 27 plugins in 9.6 sec
```
So waiting for each plugin install will not help, we actually need to wait for the last line.

Reply:
@uhafner, ignore what I said. I was under an impression this is a one-per-plugin logging message and that we are only waiting for the first one.